### DISCUSSION TITLE
Query returns zero results with 'fetch'

### DISCUSSION THREAD

**Discussion by ghost** [jeremyevans/sequel#1822] [General]
I've got a psql 14 instance with a [sample](https://www.postgresqltutorial.com/postgresql-sample-database/) database. When querying the following directly to the server (from psql) it returns the expected results:

```
select * from actor 
    where updated_at::timestamptz > '2022-01-11 11:47:08 -0300' 
    and actor_id not in ( 
        select actor_id from actor 
        where created_at::timestamptz > '2022-01-11 11:47:08 -0300'
    );
```
But from my controller on Rails (6.1.4) (Ruby 2.7.2) using:

```
def query_sample_method
    s = Sequel.connect(query_string)
    q = "select * from actor 
        where updated_at::timestamptz > '2022-01-11 11:47:08 -0300' 
        and actor_id not in ( 
            select actor_id from actor 
            where created_at::timestamptz > '2022-01-11 11:47:08 -0300'
        );"
    results = s.fetch(q).all
end
```

I get zero results.

I've also tried: `results = s[q].all`, `results = s[q]` and 
```
s.fetch(q) do |row|
    #do something
end
```
Or adding `::timestamptz` to my datetime string

(?)

**Comment by jeremyevans** [comment]
You should try using the `pg`  driver directly and see what the results are.  If you get results using `pg` directly and not with Sequel, then it is probably Sequel related.  If you don't get results using the `pg` driver directly, the issue is probably unrelated to Sequel.

**Reply by ghost** [reply]
Right. This works:

```
con = PG.connect(dbname: 'dvd_db', user: 'postgres')

rs = con.exec("select * from actor
    where updated_at::timestamptz > '2022-01-11 11:47:08 -0300'
    and actor_id not in (
        select actor_id from actor
        where created_at::timestamptz > '2022-01-11 11:47:08 -0300');"
)

rs.each do |row|
    puts row
end
```

Using `pg` v1.2.3

**Comment by jeremyevans** [comment]
Weird.  I cannot see what would be different in this case.  Sequel does set `SET DateStyle = 'ISO'` on the connection.  Does that affect results when using `pg` directly (e.g.  `con.exec("SET DateStyle = 'ISO'")` after the `PG.connect`)?

Is the connection string you are passing to `Sequel.connect` something like `postgres:///dvd_db?user=postgres`?

Is it possible for you to put together a self contained example showing this issue?
